samples: nrf9160: https_client: Option to use TF-M and MbedTLS (#5627)
nrfconnect/sdk-nrf: samples/nrf9160/https_client/README.rst

To connect to other servers, you might need to provision a different certificate.
See :ref:`cert_dwload` for more information.

> Using MbedTLS and TF-M
> **********************

Reviewer: ```suggestion
Using Mbed TLS and TF-M
***********************
```

> This sample supports using MbedTLS and Trusted Firmware-M (TF-M).

Reviewer: ```suggestion
This sample supports using Mbed TLS and Trusted Firmware-M (TF-M).
```

> Instead of offloading the TLS sockets in to the modem, the MbedTLS library from Zephyr can be utilized.

Reviewer: ```suggestion
Instead of offloading the TLS sockets into the modem, you can use the Mbed TLS library from Zephyr.
```

> Using the Zephyr MbedTLS, the offloaded sockets utilizing the modem can still be used. The MbedTLS offers more
> configuration options than is possible when using the offloaded TLS handling.

Reviewer: ```suggestion
Using the Zephyr Mbed TLS, you can still use the offloaded sockets.
Mbed TLS offers more configuration options than using the offloaded TLS handling.
```

> This sample uses the TF-M while using the MbedTLS, thus all the crypto operations are run on the secure side of the device.

Reviewer: ```suggestion
This sample uses TF-M and Mbed TLS.
Thus, all the crypto operations are run on the secure side of the device.
```

Author: In this section the TF-M is the prerequisite to run the crypto on the secure side, otherwise the crypto functionality of the MbedTLS would be used and ran on the unsecure side. Needs maybe a bit clarification here.

Author: Maybe this would be better
"While using TF-M and Mbed TLS using PSA crypto, all the crypto operations are run on the secure side on the device."

Reviewer: Using x using y sounds a bit weird. How about:
When using TF-M and Mbed TLS with PSA crypto, ....?

.. include:: /includes/build_and_run_nrf9160.txt

> To build the sample with MbedTLS and TF-M, add following to your west build-command: `-DOVERLAY_CONFIG=overlay-tfm_mbedtls.conf`

Reviewer: ```suggestion
To build the sample with Mbed TLS and TF-M, add the following to your west build command:

.. code-block:: none

   -DOVERLAY_CONFIG=overlay-tfm_mbedtls.conf
```

In addition, it uses the following sample:

* :ref:`secure_partition_manager`

> This sample offers also possibility to use the TF-M module that can be found in the following location in the |NCS| folder structure:
> 
> * ``modules/tee/tfm/``

Reviewer: ```suggestion
This sample offers also a possibility to use the TF-M module that is at :file:`modules/tee/tfm/` in the the |NCS| folder structure.
```
